From 6bbff155485af7aefd34133f2e2fa70c3f6e785b Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Mon, 4 Nov 2024 18:26:27 +0100
Subject: [PATCH] Fix reproducible builds

Do not generate absolute file names into headers

Upstream-Status: Backport [https://invent.kde.org/libraries/kopeninghours/-/merge_requests/116]

SPDX-FileCopyrightText: Andreas Cord-Landwehr <cordlandwehr@kde.org>
SPDX-License-Identifier: MIT
Signed-off-by: Andreas Cord-Landwehr <cordlandwehr@kde.org>
---
 src/lib/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lib/CMakeLists.txt b/src/lib/CMakeLists.txt
index 920f8b0..a05f11e 100644
--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -6,12 +6,13 @@ flex_target(openinghoursscanner
     openinghourslexer.l
     ${CMAKE_CURRENT_BINARY_DIR}/openinghourslexer.cpp
     DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/openinghoursscanner_p.h
-    COMPILE_FLAGS "--nounistd"
+    COMPILE_FLAGS "--nounistd --noline"
 )
 bison_target(openinghoursparser
     openinghoursparser.y
     ${CMAKE_CURRENT_BINARY_DIR}/openinghoursparser.cpp
     DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/openinghoursparser_p.h
+    COMPILE_FLAGS "--no-lines"
 )
 add_flex_bison_dependency(openinghoursscanner openinghoursparser)
 set_source_files_properties(${BISON_openinghoursparser_OUTPUTS} ${FLEX_openinghoursscanner_OUTPUTS} PROPERTIES SKIP_AUTOMOC ON)
-- 
2.43.0

